src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -521,13 +521,19 @@ SyncManager::~SyncManager()
     }
 #endif
 
-    for (auto& user : m_users) {
-        user->detach_from_sync_manager();
+    {
+        util::CheckedLockGuard lk(m_user_mutex);
+        for (auto& user : m_users) {
+            user->detach_from_sync_manager();
```

Comment:
We don't drain the vector of users when we detach them, so such check would fail. We could do it, but then we also need to make a new copy of the vector (which may be turned into a map for fast removal). It seems a bit of an overkill.